railgun/test/integration/suite_test.go

```diff
@@ -111,67 +125,95 @@ func TestMain(m *testing.M) {
-		os.Exit(25)
+		fmt.Fprintf(os.Stderr, "Error: could not create testing environment: %s", err)
+		os.Exit(ExitCodeCantCreateCluster)
 	}
 	fmt.Printf(
 		"INFO: environment built CLUSTER_NAME=(%s) CLUSTER_TYPE=(%s) ADDONS=(metallb, kong)\n",
 		env.Cluster().Name(), env.Cluster().Type(),
 	)
 	defer func() {
 		if err := env.Cleanup(ctx); err != nil {
-			fmt.Fprintf(os.Stderr, "Error: could not cleanup testing environment: %s", err.Error())
-			os.Exit(30)
+			fmt.Fprintf(os.Stderr, "Error: could not cleanup testing environment: %s", err)
+			os.Exit(ExitCodeCleanupFailed)
 		}
 	}()
 
 	fmt.Printf("INFO: waiting for cluster %s and all addons to become ready\n", env.Cluster().Name())
 	if err := <-env.WaitForReady(ctx); err != nil {
-		fmt.Fprintf(os.Stderr, "Error: testing environment never became ready: %s", err.Error())
-		os.Exit(26)
+		fmt.Fprintf(os.Stderr, "Error: testing environment never became ready: %s", err)
+		os.Exit(ExitCodeCantCreateCluster)
 	}
 
 	fmt.Printf("INFO: collecting Kong Proxy URLs from cluster %s for tests to make HTTP calls\n", env.Cluster().Name())
 	proxyURL, err = kongAddon.ProxyURL(ctx, env.Cluster())
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "Error: could not get proxy URL from Kong Addon: %s", err.Error())
-		os.Exit(27)
+		fmt.Fprintf(os.Stderr, "Error: could not get proxy URL from Kong Addon: %s", err)
+		os.Exit(ExitCodeCantCreateCluster)
 	}
 	proxyAdminURL, err = kongAddon.ProxyAdminURL(ctx, env.Cluster())
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "Error: could not get proxy URL from Kong Addon: %s", err.Error())
-		os.Exit(28)
+		fmt.Fprintf(os.Stderr, "Error: could not get proxy URL from Kong Addon: %s", err)
+		os.Exit(ExitCodeCantCreateCluster)
 	}
 	proxyUDPURL, err = kongAddon.ProxyUDPURL(ctx, env.Cluster())
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "Error: could not get proxy URL from Kong Addon: %s", err.Error())
-		os.Exit(29)
+		fmt.Fprintf(os.Stderr, "Error: could not get proxy URL from Kong Addon: %s", err)
+		os.Exit(ExitCodeCantCreateCluster)
 	}
 
 	if v := os.Getenv("KONG_BRING_MY_OWN_KIC"); v == "true" {
 		fmt.Println("WARNING: caller indicated that they will manage their own controller")
 	} else {
 		fmt.Println("INFO: deploying controller manager")
 		if err := deployControllers(ctx, controllerNamespace); err != nil {
-			fmt.Fprintln(os.Stderr, err.Error())
-			os.Exit(13)
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(ExitCodeCantCreateCluster)
 		}
 	}
 
-	fmt.Println("INFO: testing environment is ready, running tests")
+	fmt.Printf("INFO: running final testing environment checks")
+	serverVersion, err := env.Cluster().Client().ServerVersion()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error: could not retrieve server version for cluster: %s", err)
+		os.Exit(ExitCodeCantCreateCluster)
+	}
+
+	fmt.Printf("INFO: testing environment is ready KUBERNETES_VERSION=(%s): running tests\n", serverVersion.String())
```

Comment:
nit: if format printing a type implementing `fmt.Stringer`, use `%v` on the type directly instead of `%s` on `x.String()`

Reply:
```suggestion
	fmt.Printf("INFO: testing environment is ready KUBERNETES_VERSION=(%v): running tests\n", serverVersion)
```